### PR TITLE
address some int/double/num issues

### PR DESCRIPTION
**Pull request by devoncarew**
- address some int/double/num issues
- add a (commented out) section to the analysis options file, to make it easier to toggle on and off no implicit casts

### CODE REVIEW

File: packages/devtools/lib/src/utils.dart [hunk 1]
```diff
@@ -66,14 +66,14 @@ String msText(
...
   }
-  return min(a, b);
+  return min<T>(a, b);
 }
 
-num nullSafeMax(num a, num b) {
+T nullSafeMax<T extends num>(T a, T b) {
   if (a == null || b == null) {
     return a ?? b;
   }
```
Comment (kenzieschmoll) [inline packages/devtools/lib/src/utils.dart]: Do you need to add <T> to the line below this?
`max<T>(a,b);`

**Comment by kenzieschmoll** [comment]
General question: when should we be using num vs double?

File: packages/devtools/lib/src/utils.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools/lib/src/utils.dart]: Will do

**Comment by devoncarew** [comment]
I've been putting in `num` when we don't know what we'll get from some JS APIs (an int or a double). I've then let that flow through to other APIs.

Many of these changes were informed by the flag to disable implicit casts; so, it catches when you try and use a `num` as it were an `int` (but it turns out you might have gotten a `double` from a JS API).

**Pull request merged**